Src/Controller/MonitorController.hs

```diff
@@ -4,22 +4,38 @@ module Src.Controller.MonitorController where
 
     adicionaMonitor :: IO()
     adicionaMonitor = do
+        id <- insereMatricula
+        if id == 0
+            then putStrLn "Cadastro cancelado.\n"
+        else do 
+            disciplina <- insereDisciplina
+            if disciplina == "VOLTAR"
```

Comment:
Acredito que só esse ponto de retorno para o menu principal já está OK, não precisa de dois com base no id e na disciplina. Como já existe o conceito do input "VOLTAR" para voltar ao menu principal em outro trecho do sistema, seria interessante manter esse e remover essa funcionalidade de voltar quando id == 0

Reply:
Mas quando você estiver no sistema e não lembrar do matrícula do aluno? e querer voltar pra cancelar e consultar a matricula.

Reply:
o insereDisciplina pode receber o id como getLine, analisando primeiramente se a string recebido é igual a "VOLTAR". se for, volta ao menu principal, caso contrário, segue adiante no processo

Reply:
Daí basta explicar isso no menu de interação, evitando precisar pensar em dois pontos de retorno ao menu

Reply:
No caso o id é um Int, por isso coloquei == 0, mas eu posso usar essa lógica. Passo o ID para dentro de insere disciplina, se for igual a zero ele volta ao menu principal, o que acha?

Reply:
Pode ser. Se quiser usar a lógica do VOLTAR também funciona, recebendo o id primeiramente como string, tratando o caso de ser VOLTAR e depois convertendo para Int em uma outra constante para seguir adiante. Gosto de pensar assim pq é a forma que o sistema usa em outros pontos para informar ao usuário sobre a possibilidade de voltar ao menu principal.

Mas dessa forma também ta legal, pode fazer assim